lib/iris/cube.py

```diff
@@ -716,9 +714,8 @@ def __init__(self, data, standard_name=None, long_name=None,
         if isinstance(data, six.string_types):
             raise TypeError('Invalid data type: {!r}.'.format(data))
 
-        self._dask_array = None
-        self._numpy_array = None
-        self.data = data
+        # Initialise the cube data manager.
+        self._data_manager = DataManager(data, realised_dtype=dtype)
```

Comment:
The `realised_dtype` is used by the `DataManager` to realise the lazy, masked integral/bool data case, where the original `dtype` (that is integral or bool) has been converted to float in order to handle the masking with `nan`s in `dask` ... does that help?